File: ios/TmuxWebClient/Models/Project.swift
import Foundation

/// Mirrors `Project` in ../../src/projects.ts
struct Project: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let repoPath: String
    let createdAt: String
}

struct ProjectListResponse: Codable {
    let projects: [Project]
}

struct NewProjectRequest: Codable {
    let name: String
    let repoPath: String
}

File: kmp/composeApp/src/commonMain/kotlin/com/tanyudii/tmuxweb/domain/model/Project.kt
package com.tanyudii.tmuxweb.domain.model

import kotlinx.serialization.Serializable

/** Mirrors `Project` in src/projects.ts (backend contract, frozen — see the plan §2.4). */
@Serializable
data class Project(
    val id: String,
    val name: String,
    val repoPath: String,
    val createdAt: String,
)

@Serializable
data class ProjectListResponse(val projects: List<Project>)

@Serializable
data class NewProjectRequest(val name: String, val repoPath: String)
